### PR TITLE
Publish NuGet Package to Internal Office Feed

### PR DESCRIPTION
**Pull request by samuelfreiberg**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [X] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

In order to add test coverage for FURN in Devmain, we need to have the ability to pull the FluentTester bundle into Devmain. To achieve this goal, we need to publish the JS bundle to an internal Office feed as a NuGet package. Then, we can pull it in via Devmain.

Specifically, we are publishing it to this [React-Native Office Feed](https://dev.azure.com/ms/react-native/_packaging?_a=feed&feed=react-native). This is a private feed for Office (no public access), which is desired for us because this is for internal-testing purposes only.

### Verification

(how the change was tested, including both manual and automated tests)

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: .ado/templates/win32-nuget-publish.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    workingDirectory: apps/win32
+    displayName: 'Bundling win32'
+
+  # Pack the NuGet package
+  - task: CmdLine@1
+    displayName: 'Create NuGet package for Win32 FluentTester bundle.'
+    inputs:
+        filename: nuget
+        arguments: 'pack Microsoft.FluentUI.FluentTesterWin32.nuspec -OutputDirectory $(Build.ArtifactStagingDirectory)/nuget -OutputFileNamesWithoutVersion -Verbosity detailed -Version $(Build.BuildNumber)'
+        workingFolder: 'apps/win32/nuget'
```
Comment (Saadnajmi) [inline .ado/templates/win32-nuget-publish.yml:18]: Do you have to create this folder beforehand or anything? The apple NuGet makes a `DerivedData` intermediate folder off of root, and I think that works fine without creating the folder beforehand.

**Review by Saadnajmi — APPROVED** [review]

File: apps/win32/nuget/Microsoft.FluentUI.FluentTesterWin32.nuspec [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd">
+  <metadata>
+    <id>Microsoft.FluentUI.FluentTesterWin32</id>
+    <version>1.0.0</version>
```
Comment (Saadnajmi) [inline apps/win32/nuget/Microsoft.FluentUI.FluentTesterWin32.nuspec:5]: We have a few scripts that set the version number to "Date-<number of publishes today>" that you might be able to reuse

File: apps/win32/nuget/Microsoft.FluentUI.FluentTesterWin32.nuspec [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    <title>FluentUI-React-Native Win32 Bundle</title>
+    <authors>Microsoft Office CXE</authors>
+    <projectUrl>https://github.com/microsoft/fluentui-react-native.git</projectUrl>
+    <description>This package contains the React Native JS bundle of FluentUI-React-Native's Win32 Test App.</description>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+  </metadata>
+  <files>
+    <file src="..\dist\index.win32.bundle" target="win32" />
```
Comment (Saadnajmi) [inline apps/win32/nuget/Microsoft.FluentUI.FluentTesterWin32.nuspec:13]: Generally things like `.nuspec` / `.podspec` / `package.json` files are at the root of the repo, but I suppose this is fine?

Comment (samuelfreiberg) [reply apps/win32/nuget/Microsoft.FluentUI.FluentTesterWin32.nuspec:13]: I looked at a bunch of examples in fabric-internal and other standalone feature repos. I found they were usually in specific directories. I'm assuming both ways are ok :)

**Comment by samuelfreiberg** [comment]
> If you want to test this change, what I've done previously is either:
> 
> * Make a branch off of the repo (not your fork) and manually create a pipeline & run it
> * Put the yaml job in the PR pipeline temporarily, see that everything worked, and then move it back to Publish only
> 
> It's not perfect, but giving my experience from making the other NuGet in this repo~

This is extremely helpful - I was trying to figure out how to test it. Thanks!

File: apps/win32/nuget/Microsoft.FluentUI.FluentTesterWin32.nuspec (hunk 3 above)
Comment (Saadnajmi) [reply apps/win32/nuget/Microsoft.FluentUI.FluentTesterWin32.nuspec:13]: Ah, good point! Forgot about fabric-internal 😆

**Pull request closed without merging**